fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/erase/EraseTestMain.java

```diff
@@ -0,0 +1,211 @@
+                    c.rollback();
+                    throw x;
+                }
+                c.commit();
+            }
+        } catch (SQLException x) {
+            throw translator.translate(x);
+        }
+    }
+
+    /**
+     * determines the flavor and schema of the execution.
+     */
+    public void determineFlavorAnSchema() {
+        if (inProperties.getProperty("type") != null && inProperties.getProperty("type").equals("db2")) {
+            dbType = DbType.DB2;
+            if (inProperties.getProperty("db2.schemaName") != null) {
+                schemaName = inProperties.getProperty("db2.schemaName");
+            }
+        } else {
+            if (inProperties.getProperty("postgres.schemaName") != null) {
+                schemaName = inProperties.getProperty("db2.schemaName");
+            }
+        }
+    }
+
+    /**
+     * process args
+     */
+    public void processArgs() {
+        String prefix = "db2.";
+        if (dbType == DbType.POSTGRESQL) {
+            prefix = "postgres.";
+        }
+        dbProperties = new Properties();
+        for(Entry<Object, Object> prop : inProperties.entrySet()) {
+            String key = (String) prop.getKey();
+            if (key.startsWith(prefix)) {
+                dbProperties.put(key.replace(prefix, ""), prop.getValue());
+            }
+        }
+        resourceType = args[0];
+        logicalId = args[1];
+    }
+
+    /**
+     * @param args
+     * @throws Exception
+     */
+    public static void main(String[] args) throws Exception {
+        Properties properties = new Properties();
+        properties.load(new FileInputStream("src/test/resources/test.erase.properties"));
+        EraseTestMain main = new EraseTestMain(properties, args);
+        main.determineFlavorAnSchema();
+        main.processArgs();
+        main.erase();
+    }
+
+    protected Connection createConnection() {
+        Properties connectionProperties = new Properties();
+        JdbcPropertyAdapter adapter = getPropertyAdapter(dbType, dbProperties);
+        adapter.getExtraProperties(connectionProperties);
+
+        String url = translator.getUrl(dbProperties);
+        logger.info("Opening connection to: " + url);
+        Connection connection;
+        try {
+            connection = DriverManager.getConnection(url, connectionProperties);
+            connection.setAutoCommit(false);
+        } catch (SQLException x) {
+            throw translator.translate(x);
+        }
+        return connection;
+    }
+
+    /**
+     * Load the driver class
+     */
+    public static void loadDriver(IDatabaseTranslator translator) {
+        try {
+            Class.forName(translator.getDriverClassName());
+        } catch (ClassNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    public static JdbcPropertyAdapter getPropertyAdapter(DbType dbType, Properties props) {
```

Comment:
public method - comment.

Reply:
Turns out we have a common method. getPropertyAdapter in CommonUtil